go/integTest/shared_commands_test.go

```diff
@@ -4602,6 +4602,12 @@ func (suite *GlideTestSuite) TestXRead() {
 		case <-finished:
 			assert.Fail(suite.T(), "Infinite block finished")
 		case <-time.After(3 * time.Second):
+			_, err := client.XAdd(
+				key1,
+				[][]string{{"k1_field1", "k1_value1"}, {"k1_field1", "k1_value2"}},
+			)
+			assert.Nil(suite.T(), err)
+			<-finished
```

Comment:
The above `testClient.XReadWithOptions` needs to be finished to close the `testClient` below.